rcl/include/rcl/time.h

```diff
@@ -54,12 +54,67 @@ typedef enum rcl_clock_type_t
   RCL_STEADY_TIME
 } rcl_clock_type_t;
 
+/// A duration of time, measured in nanoseconds and its source.
+typedef struct rcl_duration_t
+{
+  rcl_duration_value_t nanoseconds;
+} rcl_duration_t;
+
+/// Enumeration to describe the type of time jump.
+typedef enum rcl_clock_change_t
+{
+  /// The source before and after the jump is ROS_TIME.
+  RCL_ROS_TIME_NO_CHANGE = 1,
+  /// The source switched to ROS_TIME from SYSTEM_TIME.
+  RCL_ROS_TIME_ACTIVATED = 2,
+  /// The source switched to SYSTEM_TIME from ROS_TIME.
+  RCL_ROS_TIME_DEACTIVATED = 3,
+  /// The source before and after the jump is SYSTEM_TIME.
+  RCL_SYSTEM_TIME_NO_CHANGE = 4
+} rcl_clock_change_t;
```

Comment:
Maybe - sounds reasonable to me.

It would be good if @tfoote could provide some context why this API / approach was chosen in `rclcpp`. Maybe there is something we are missing.